route-rs-runtime/src/link/primitive/join_link.rs

```diff
@@ -208,35 +216,37 @@ impl<Packet: Sized> Drop for JoinEgressor<Packet> {
 
 impl<Packet: Sized> Stream for JoinEgressor<Packet> {
     type Item = Packet;
-    type Error = ();
 
     /// Iterate over all the channels, pull the first packet that is available.
     /// This starts at the next index after the last successful recv
-    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
```

Comment:
Comment: I like that `Future`s and `Stream`s have different poll fns `poll / poll_next` to help distinguish them.